### PR TITLE
Make create_dm_room public

### PR DESCRIPTION
**Pull request by poljar**
This replaces #961.

It's just removing some dead code and making fmt/clippy happy.

### CODE REVIEW

File: crates/matrix-sdk/src/client/mod.rs [hunk 1]
```diff
@@ -1846,6 +1847,44 @@ impl Client {
...
+        // Now we need to mark the room as a DM for ourselves, we fetch the
+        // existing `m.direct` event and append the room to the list of DMs we
+        // have with this user.
+        let mut content = self
+            .store()
+            .get_account_data_event(GlobalAccountDataEventType::Direct)
+            .await?
+            .map(|e| e.deserialize_as::<DirectEvent>())
+            .transpose()?
+            .map(|e| e.content)
```
Comment (jplatte) [inline crates/matrix-sdk/src/client/mod.rs:1875]:
```suggestion
            .account()
            .account_data::<DirectEventContent>()
            .await?
            .map(|c| c.deserialize())
            .transpose()?
```

**Comment by jplatte** [comment]
I probably should have stated that I didn't look up whether what I suggested above was exactly right 😅

**Review by jplatte — APPROVED** [review]
… but if CI succeeds, I think we can merge.

File: crates/matrix-sdk/src/client/mod.rs [hunk 2]
```diff
@@ -1872,7 +1874,7 @@ impl Client {
             .await?
             .map(|c| c.deserialize())
             .transpose()?
-            .unwrap_or_else(|| ruma::events::direct::DirectEventContent(BTreeMap::new()));
+            .unwrap_or_else(|| DirectEventContent(BTreeMap::new()));
```
Comment (jplatte) [inline crates/matrix-sdk/src/client/mod.rs:1877]: Actually, with the current Ruma rev, this can be:

```suggestion
            .unwrap_or_default();
```

**Comment by poljar** [comment]
I tested it manually, so it at least worked at the time of testing.

Thanks for the initial patch.

**Pull request merged**